src/lib/blkdata_svc/blkdata_service.cpp

```diff
@@ -43,14 +45,15 @@ void BlkDataService::create_vdev(uint64_t size) {
                                                         .num_chunks = 1,
                                                         .blk_size = phys_page_size,
                                                         .dev_type = HSDevType::Data,
+                                                        .alloc_type = alloc_type,
+                                                        .chunk_sel_type = chunk_sel_type,
```

Comment:
I think blk_alloc_type is good, but suggest chunk selector instance needs to be driven by the callbacks as part of homestore::init(). 

So I was making changes something like this
```
bool HomeStore::start(const hs_input_params& input, hs_before_services_starting_cb_t svcs_starting_cb,
                      std::unique_ptr< IndexServiceCallbacks > index_cbs,
                      std::unique_ptr< ReplServiceCallbacks > repl_cbs)
```
so this callbacks will have a method create_chunk_selector() which will be invoked. Otherwise HomeObject will not have a chance to create its instance with its own constructor right?

Reply:
HO will initialize homestore with `HomeStore::instance()->format_and_start`, which will takes `hs_format_params` as input and in which HO can specify for which service it want which type of allocator/chunkselector, right?

Reply:
@JacksonYao287 HO just indicate to HomeStore that dataservice want heapchunkselector (which is a ENUM value), and homestore will initialize its chunkselector type to HeapChunkSelector.

Reply:
@yamingk  the question is that we do not have  the implemention of HeapChunkSelector inside homestore, it is implement as a derived class of `ChunkSelector` in Homeobject, so Homestore can not get the defination of HeapChunkSelector for itself. that why i want to clarify how to pass HeapChunkSelector from homeobject to homestore